client/src/pages/Profile/EditProfile/EditProfile.tsx

```diff
@@ -1,15 +1,321 @@
-import Box from '@material-ui/core/Box';
-import Grid from '@material-ui/core/Grid';
-import Paper from '@material-ui/core/Paper';
-import Typography from '@material-ui/core/Typography';
+import { useState, useEffect } from 'react';
+import {
+  Button,
+  TextField,
+  Grid,
+  FormControl,
+  MenuItem,
+  Select,
+  Typography,
+  Paper,
+  Box,
+  Switch,
+} from '@material-ui/core';
+import useStyles from './useStyles';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { useAuth } from '../../../context/useAuthContext';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import { Gender, Profile } from '../../../interface/Profile';
+import { createProfile, getMyProfile, updateProfile } from '../../../helpers/APICalls/profiles';
+import { useHistory } from 'react-router-dom';
+
+const validationSchema = yup.object({
+  firstName: yup.string().required('First Name is required'),
+  lastName: yup.string().required('Last Name is required'),
+  gender: yup.string().required('Gender is required'),
+  birth: yup.string().required('Birth is required'),
+  phone: yup.string().required('Phone is required'),
+  address: yup.string().required('Address is required'),
+  description: yup.string().required('Your description is required'),
+});
 
 const EditProfile = (): JSX.Element => {
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+  const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
+  const [profileId, setProfileId] = useState('');
+  const [fetchedProfile, setFetchedProfile] = useState({
+    userId: loggedInUser?.id,
+    firstName: '',
+    lastName: '',
+    gender: 'Male' as Gender,
+    birth: '',
+    phone: '',
+    address: '',
+    description: '',
+    isSitter: false,
+    title: '',
+    hourlyWage: 0,
+  });
+
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  useEffect(() => {
+    try {
+      const fetchProfiles = async () => {
+        const myProfile = await getMyProfile();
+        if (myProfile._id) {
+          setFetchedProfile(myProfile);
+          setProfileId(myProfile._id);
+          setIsEditMode(true);
+        }
+      };
+      fetchProfiles();
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        updateSnackBarMessage(error.message);
+      }
+    }
+  }, [updateSnackBarMessage]);
+
+  const handleSubmit = async (values: {
+    userId: string | undefined;
+    firstName: string;
+    lastName: string;
+    gender: Gender;
+    birth: string;
+    phone: string;
+    address: string;
+    description: string;
+    isSitter: boolean;
+    title: string;
+    hourlyWage: number;
+  }) => {
+    if (isEditMode) {
+      await updateProfile(profileId, values as Profile);
```

Comment:
It would be nice to update the snackbar in case of success/failure :)

Reply:
Applied